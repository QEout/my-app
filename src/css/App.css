.App {
  text-align: center;
  width: 100vw;
  height: 100vh;
  display: flex;
  font-size: 32px;
  color: #3498db;
  position: relative;
  align-items: center;
  justify-content: center;
}

.main {
  position: relative;
}

.title {
  font-size: 36px;
  font-weight: bold;
  padding: 40px;
  letter-spacing: 6px;
}

.result {
  height: 240px;
}

.error {
  text-align: left;
  margin-left: 78px;
}

.input {
  border-left: none;
  border-right: none;
  border-top: none;
  font-size: inherit;
  border-bottom: 2px solid #3498db;
}

.input:focus {
  outline: none;
}

.userCard {
  display: inline-flex;
  border: 2px solid #3498db;
  border-radius: 10px;
  margin: 40px;
  padding: 20px;
  background-color: #F5F5F6;
}

.userCard img {
  width: 100px;
  height: 100px;
  border-radius: 10%;
  margin-right: 20px;
}

.userCard .info {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.qq {
  color: grey
}