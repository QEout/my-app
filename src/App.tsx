import React, { useState } from "react";
import "./css/App.css";
import "./css/loading.css"
// 使用lodash
import _ from "lodash";
import { get } from './utils/request'

type QInfo = {
  name: string;
  qlogo: string;
  qq: string;
};

export function isValidQQ(qq: string): boolean {
  return /^[1-9]\d{4,11}$/.test(qq);
}

function App() {
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState<string>("");
  const [data, setData] = useState<QInfo | null>(null);

  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    handleDebounce(e.target.value.trim());
  };

  const onSearch = (value: any) => {
    if (value.length === 0) {
      setLoading(false);
      setData(null);
      setError("");
      return;
    }

    if (!isValidQQ(value)) {
      setData(null);
      setError("请输入正确的QQ号");
      return;
    }

    setError("");
    setLoading(true);
    get('https://api.uomg.com/api/qq.info', { qq: value })
      .then((res: any) => {
        if (res.code !== 1) {
          setError(res.msg);
          setData(null);
        } else {
          setError("");
          setData({
            name: res.name,
            qlogo: res.qlogo,
            qq: res.qq,
          });
        }
      }).finally(() => setLoading(false));
  };

  const handleDebounce = _.debounce(onSearch, 500);

  return (
    <div className="App">
      <div className="main">
        <div className="title">QQ号查询</div>
        <div>
          <span>QQ：</span>
          <input
            className="input"
            type="text"
            placeholder="请输入QQ号"
            onChange={onChange}
          /></div>
        <div className="result">
          {loading && <div id="preloader"></div>}
          {!loading && data && <UserComponent {...data} />}
          {error && <div className="error">{error}</div>}
        </div>
      </div>
    </div>
  );
}

// loading组件
function UserComponent(data: QInfo) {
  return (
    <div className="userCard">
      <img
        className="avatar"
        src={data?.qlogo}
        alt="avatar"
      />
      <div className="info">
        <div title={data?.name} className="name">
          {data?.name}
        </div>
        <div className="qq">{data?.qq}</div>
      </div>
    </div>)
}
export default App;
