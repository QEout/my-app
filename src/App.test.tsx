import App, { isValidQQ } from "./App";
import renderer from "react-test-renderer";

test("valid input", () => {
	expect(isValidQQ("123456789")).toBe(true);
	expect(isValidQQ("a123456789")).toBe(false);
	expect(isValidQQ("01234")).toBe(false);
	expect(isValidQQ("abc")).toBe(false);
	expect(isValidQQ("  ")).toBe(false);
});

test("snapshot", () => {
	const tree: any = renderer.create(<App />).toJSON();
	expect(tree).toMatchSnapshot();
});

test("fetch with wrong qq value", async () => {
	const data = await fetch(`https://api.uomg.com/api/qq.info?qq=1`).then(
		(res) => res.json()
	);
	expect(data.name).toBe(null);
});

test("fetch with right qq value", async () => {
	const data = await fetch(`https://api.uomg.com/api/qq.info?qq=123456`).then(
		(res) => res.json()
	);
  expect(data.name).not.toBe(null);
	expect(data.qq).toBe("123456");
});
