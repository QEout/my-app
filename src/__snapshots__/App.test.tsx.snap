// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="App"
>
  <div
    className="main"
  >
    <div
      className="title"
    >
      QQ号查询
    </div>
    <div>
      <span>
        QQ：
      </span>
      <input
        className="input"
        onChange={[Function]}
        placeholder="请输入QQ号"
        type="text"
      />
    </div>
    <div
      className="result"
    />
  </div>
</div>
`;
